src/emulation/mod.rs

```diff
@@ -137,18 +139,47 @@ impl SyscallMock {
     }
 }
 
+pub enum Executable {
+    Interpreter(PathBuf),
+    Program(PathBuf),
+}
```

Comment:
This enum exists to differentiate between an executable that refers to the interpreter that should be invoked, i.e. `ruby`, and a program that is itself executable.